azure_li_services/schema.py

```diff
@@ -108,6 +108,20 @@
                     'required': False,
                     'type': 'string'
                 },
+                'ssh-private-key': {
+                    'required': False,
```

Comment:
We will want some runtime check. Since storage is optional obviously this also needs to be optional. However, if storage is configure in the YAML we will also need to have the private key, so after the schema validation we need to have a conditional runtime check.

Reply:
But the storage configuration is just handling fstab entries for mount procedures. How is that related to the ssh setup of users(credentials section) ?

So far we have not specified relations between sections of the yaml. This type of information is normally done by schematron rules which does not exist in the yaml markup. So I agree if we want to have that a runtime checker in python code would be the best option. However I suggest to handle this in an extra issue